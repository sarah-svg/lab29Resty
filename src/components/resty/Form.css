.Form{
    margin-right: .3vw;
    width: 81.9vw;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  
    div{
      display: flex;
      width: 100%;
      justify-content: flex-start;
      align-items: center;
      flex-direction: row;
  
      input[type="radio"]{
        
        visibility: hidden;
        height: 0;
        width: 0;
        margin-left: 0; 
        background-color: teal;
      }
    
      label{
        color: rgb(18, 173, 148);
        padding: 4px 16px;
        cursor: pointer;
      }
  
      input[type="radio"]:checked+label {
        background-color: rgb(60, 209, 209);
        color: rgb(38, 132, 187);
        
      }
      
    
    }
  
    input{
      
      margin-bottom: 8px;
      align-items: center;
     
    }
  
    button{
      margin-left: 3vw;
      background-color: teal;
    }
  
    textarea{
      background-color:aquamarine;
      margin-top: 5px;
      margin-left: -15%;
      width: 80%;
      height: 6vw;
      align-self: center;
    }
  }
  
  .url{
    
    background-color: rgb(48, 184, 165);
    width: 80%;
    height: 25px;
    font-size: 12px;
    margin-right: 15%;
    margin-top: 5px;
  }