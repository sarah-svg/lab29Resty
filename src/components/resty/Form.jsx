import React from 'react';
import PropTypes from 'prop-types';
import styles from './Form.css';

const Form = ({ url, method, body, onSubmit, onChange }) => {
  return (  

    <div>
      <form onSubmit={onSubmit} className={styles.Form}>
        <input type="text" className={styles.url} name="url" 
          placeholder="Place URL here..." value={url} onChange={onChange} />
        <div>
          <section className="bouncy">
            <input type="radio" id="get" name="method" value="GET" 
              checked={method === 'GET'} onChange={onChange} />
            <label htmlFor="get">GET</label>

            <input type="radio" id="post" name="method" value="POST" 
              checked={method === 'POST'} onChange={onChange} />
            <label htmlFor="post">POST</label>

            <input type="radio" id="put" name="method" value="PUT" 
              checked={method === 'PUT'} onChange={onChange} />
            <label htmlFor="put">PUT</label>

            <input type="radio" id="patch" name="method" value="PATCH" 
              checked={method === 'PATCH'} onChange={onChange} />
            <label htmlFor="patch">PATCH</label>

            <input type="radio" id="delete" name="method" value="DELETE"
              checked={method === 'DELETE'} onChange={onChange} />
            <label htmlFor="delete">DELETE</label>
          </section>
          <button>Resty</button>

        </div>
        <textarea placeholder="JSON in the raw" name="body" 
          value={body} onChange={onChange}></textarea>
      </form>
            
    </div>
  );
};

Form.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired

};

export default Form;

