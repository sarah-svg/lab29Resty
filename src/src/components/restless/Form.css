.Form{
  margin-right: .3vw;
  width: 81.9vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  section{
    display: flex;
    width: 100%;
    justify-content: flex-start;
    align-items: center;
    flex-direction: row;

    input[type="radio"]{
      visibility: hidden;
      height: 0;
      width: 0;
      margin-left: 0;
    }
  
    label{
      padding: 4px 16px;
      cursor: pointer;
    }

    input[type="radio"]:checked+label{
      background-color: #204060;
      color: white;
    }
  }

  input{
    margin-bottom: 8px;
  }

  button{
    margin-left: 2vw;
  }

  textarea{
    margin-top: 5px;
    width: 50%;
    height: 6vw;
    align-self: flex-start;
  }
}

.url{
  width: 100%;
  height: 25px;
  font-size: 12px;
  margin-top: 5px;
}