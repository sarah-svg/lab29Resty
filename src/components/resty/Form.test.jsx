import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Form from './Form';

describe('displays radio buttons with methods', () => {
  afterEach(() => cleanup());
  it('renders data from api with radio buttons', () => {
    const { asFragment } = render(<Form url="" method="" body="" 
      onSubmit="" onChange=""  />);
    expect(asFragment()).toMatchSnapshot();
  });
});
