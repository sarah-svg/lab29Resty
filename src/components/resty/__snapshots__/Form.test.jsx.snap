// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays radio buttons with methods renders data from api with radio buttons 1`] = `
<DocumentFragment>
  <div>
    <form
      class="Form"
    >
      <input
        class="url"
        name="url"
        placeholder="Place URL here..."
        type="text"
        value=""
      />
      <div>
        <section
          class="bouncy"
        >
          <input
            id="get"
            name="method"
            type="radio"
            value="GET"
          />
          <label
            for="get"
          >
            GET
          </label>
          <input
            id="post"
            name="method"
            type="radio"
            value="POST"
          />
          <label
            for="post"
          >
            POST
          </label>
          <input
            id="put"
            name="method"
            type="radio"
            value="PUT"
          />
          <label
            for="put"
          >
            PUT
          </label>
          <input
            id="patch"
            name="method"
            type="radio"
            value="PATCH"
          />
          <label
            for="patch"
          >
            PATCH
          </label>
          <input
            id="delete"
            name="method"
            type="radio"
            value="DELETE"
          />
          <label
            for="delete"
          >
            DELETE
          </label>
        </section>
        <button>
          Resty
        </button>
      </div>
      <textarea
        name="body"
        placeholder="JSON in the raw"
      />
    </form>
  </div>
</DocumentFragment>
`;
